### PR TITLE
SNMPv3 Trap Destinations

### PR DESCRIPTION
**Pull request by SHANDCRUZ**
### Description
Added support to Appliance SNMPv3 Trap Destinations.
(pending) 
-> didn't initialize the go.mod and go.sum

### Issues Resolved
NA

### Check List
- [ ] New functionality includes testing.
  - [ ]  All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [ ] New endpoints supported are updated in the endpoints-support.md file.
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: docs/d/appliance_snmpv3_trap_destinations.html.markdown [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+## Example Usage
+
+```hcl
+data "oneview_snmpv3_trap_destinations" "test" {
+ id_field = "67003649-af34-4a92-a46a-137855ddc8f7"
+}
+
+output "oneview_snmpv3_trap_destination_value" {
+ value = "${data.oneview_snmpv3_trap_destinations.snmptrap}"
```
Comment (yuvirani) [inline docs/d/appliance_snmpv3_trap_destinations.html.markdown:21]: it should data.oneview_snmpv3_trap_destinations.test

File: oneview/provider.go [hunk 2]
```diff
@@ -69,6 +69,7 @@ func Provider() terraform.ResourceProvider {
 
 		DataSourcesMap: map[string]*schema.Resource{
 			"oneview_appliance_snmp_v1_trap_destinations": dataSourceSNMPv1TrapDestination(),
+			"oneview_snmpv3_trap_destinations":            dataSourceSNMPv3TrapDestination(),
```
Comment (yuvirani) [inline oneview/provider.go:72]: Can you change to oneview_appliance_snmpv3_trap_destinations or change the resource order.

File: oneview/provider.go [hunk 3]
```diff
@@ -101,6 +102,7 @@ func Provider() terraform.ResourceProvider {
 
 		ResourcesMap: map[string]*schema.Resource{
 			"oneview_appliance_snmp_v1_trap_destinations": resourceSNMPv1TrapDestination(),
+			"oneview_snmpv3_trap_destinations":            resourceSNMPv3TrapDestination(),
```
Comment (yuvirani) [inline oneview/provider.go:105]: same as above

File: oneview/resource_appliance_snmpv3_trap_destinations_test.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+			{
+				Config: testAccSNMPv3TrapDestinations,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSNMPv3TrapDestinationsExists(
+						"oneview_snmpv3_trap_destinations.test", &SNMPv3TrapDestinations),
+					resource.TestCheckResourceAttr(
+						"oneview_snmpv3_trap_destinations.test", "destination_address", "test_destination_address",
+					),
+					resource.TestCheckResourceAttr(
+						"oneview_snmpv3_trap_destinations.test", "user_id", "test_user_id",
```
Comment (yuvirani) [inline oneview/resource_appliance_snmpv3_trap_destinations_test.go:40]: Add testCheck for port also

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,7 +10,13 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 # [v6.2.0-13] (unreleased)
 
 #### Major Changes
-1. Added support to Connection Templates Resource.
+1. Added support to Appliance SNMPv3 Trap Destinations
```
Comment (VenkateshRavula) [inline CHANGELOG.md:13]: Also add this feature under "oneview features supported" list

**Review by VenkateshRavula — APPROVED** [review]
LGTM

**Review by yuvirani — APPROVED** [review]

**Pull request merged**